csharp/rust/src/helpers.rs

```diff
@@ -0,0 +1,104 @@
+﻿use crate::data::Utf8OrEmptyError;
+use crate::helpers;
+use std::ffi::{c_char, CString};
+use std::ptr::null;
+use std::slice::from_raw_parts;
+use std::str::{FromStr, Utf8Error};
+
+/// # Summary
+/// Convert raw C string to an optional rust string.
+///
+/// # Safety
+/// * Dereferenced [input] ptr must contain a valid ptr or null.
+pub fn grab_str(input: *const c_char) -> Result<Option<String>, Utf8Error> {
+    if input.is_null() {
+        return Ok(None);
+    }
+    unsafe {
+        let c_str = std::ffi::CStr::from_ptr(input);
```

Comment:
Same - we won't marshal `CStr`/`CString` - just `byte[]`, just hardcore

Reply:
This is primarily about ensuring panic safety. For example, errors can occur when passing the address during a connection request. Since this is a helper method, without proper handling it could cause the program to abort; particularly in cases like UTF errors resulting from invalid input. By handling these scenarios gracefully, we avoid unexpected crashes and improve the overall robustness of the code.